docs/hazmat/primitives/asymmetric/ec.rst

```diff
@@ -122,6 +122,43 @@ Elliptic Curve Signature Algorithms
         :returns: A new instance of a :class:`EllipticCurvePublicKey`
             provider.
 
+Elliptic Curve Key Exchange algorithm
+-------------------------------------
+
+.. class:: ECDH(algorithm)
```

Comment:
A propertyt of ecdhe.public_key sounds ok to me.
I am not sure about a function, the public_key is not computed at the time you request it

On agree() I also considered briefly to call it compute() as in "compute this key exchange", but it doesn't read as well, compute_key() may do though I guess.